.github/workflows/cicd.yaml

```diff
@@ -337,32 +351,50 @@ jobs:
 
   upgrade-test:
     runs-on: ubuntu-latest
-    needs: [build]
+    needs: [build,context]
+    permissions:
+      packages: read
+    env:
+      IMAGE: ${{ needs.context.outputs.build_image }}
     steps:
       - uses: actions/checkout@v3
+      - name: Login with ghcr
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
       - name: Install yq and bash
         run: |
           sudo snap install yq
           sudo apt update
           sudo apt install bash -y
         shell: bash
-      - uses: actions/download-artifact@v3
-        with:
-          name: images
-      - name: Load Docker images
-        run: |
-          docker load -i ${GITHUB_SHA}_image.tar
-        shell: bash
       - uses: ./.github/actions/k8s-version-config
         name: Setup k8s cluster
         with:
           k8s-version: v1.25
+      - name: Configure Cluster
+        run: |
+          kubectl create ns connaisseur
+          kubectl create secret generic ${imagepullsecret} \
+            --from-file=.dockerconfigjson=$HOME/.docker/config.json \
+            --type=kubernetes.io/dockerconfigjson \
+            --namespace=connaisseur
       - uses: actions/checkout@v3
         with:
           ref: "master"
+      - name: Checkout required current files
```

Comment:
If it's about current files, why does the checkout action above not suffice?

Also, why do we need a 3rd party integration for this?